File: Data Structures & Algorithms/two-integer-sum/submission-0.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ## what can enumerate do?
        ##  For each iteration, it yields a tuple in the format (index, value), which can be easily unpacked into two variables within a for loop.

        hash_map = {}

        for i, j in enumerate(nums):
            rem = target - j
            if rem in hash_map:
                return [hash_map[rem], i]

            hash_map[j] = i
        return []

File: Data Structures & Algorithms/remove-element/submission-0.cpp
class Solution {
public:
    int removeElement(vector<int>& nums, int val) {
        vector<int> temp(nums.size());
        int i = 0;
        for(auto num: nums){
            if(num!=val){
                temp[i] = num;
                i++;
            }
        }

        nums = temp;
        return i; 
    }
};